src/amplitude-client.js

```diff
@@ -76,17 +78,37 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
   }
 
   try {
+    _parseConfig(this.options, opt_config);
+
+    if (this.options.cookieName !== DEFAULT_OPTIONS.cookieName) {
+      utils.log.warn('The cookieName option is deprecated. We will be ignoring it for newer cookies');
+    }
+
     this.options.apiKey = apiKey;
-    this._storageSuffix = '_' + apiKey + this._legacyStorageSuffix;
+    this._storageSuffix = '_' + apiKey + (this._instanceName === Constants.DEFAULT_INSTANCE ? '' : '_' + this._instanceName);
+    this._storageSuffixV5 = apiKey.slice(0,6);
```

Comment:
the storage suffix v5 is only used for the new metadata storage? This logic is a little dense to read through with all the different suffixes, and keys, old cookie, new cookie. Is it possible to group the statements and add comments? Like // setting up legacy storage and keys, // setting up new storage and keys, etc etc